src/main/java/org/opensearch/knn/plugin/transport/ClearCacheAction.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin.transport;
+
+import org.opensearch.action.ActionType;
+import org.opensearch.core.common.io.stream.Writeable;
+
+/**
+ * Action associated with ClearCache
+ */
+public class ClearCacheAction extends ActionType<ClearCacheResponse> {
+
+    public static final ClearCacheAction INSTANCE = new ClearCacheAction();
+    public static final String NAME = "cluster:admin/clear_cache_action";
+
+    private ClearCacheAction() {
```

Comment:
shouldn't this be public?

Reply:
no, it is private. We will be only using it in this class to create the INSTANCE object.